OmsAgent/omsagent.py

```diff
@@ -1010,6 +1016,20 @@ def exit_if_openssl_unavailable(operation):
     return 0
 
 
+def exit_if_gpg_unavailable(operation):
+    """
+    Check if gpg is available to use
+    If not, attempt to install
+    If install fails, throw error to return UnsupportedGpg error code
+    """
+    check_exit_code, _ = run_get_output('which gpg', True, False)
+    if check_exit_code is not 0:
+        exit_code, output = run_get_output(InstallExtraPackageCommand.format('gpg'))
+        if exit_code is not 0:
+            log_and_exit(operation, UnsupportedGpg, 'GPG could not be installed: {0}'.format(output))
```

Comment:
Do you want to log an informational message prior to install that says "Installing missing dependancy gpg.." then add another message "Gpg installed." if install completes with no error?

Reply:
Added some logging statements before and after, thank you for the catch! Let me know if I should change up the wording.